Cesium.CodeGen.Tests/CodeGenMethodTests.cs

```diff
@@ -296,4 +296,18 @@ public void ImplicitReturnDisallowedNonMain() => DoesNotCompile(@"int foo()
 {
     int unused;
 }", "Function foo has no return statement.");
+
+    [Fact]
+    public Task StructSubscriptionTest() => DoTest(@"typedef struct
+{
+    int fixedArr[4];
```

Comment:
In C, this struct has not a pointer field, but a special place for 4 `int` objects embedded directly into the structure body. To specify a type for such an unusual field, we generate a special type called `<syntheticBuffer>` (an unusual name to prevent the user from accessing it in their code directly). This is more or less in line with what Roslyn does for a `fixed` array.